imap_processing/cdf/cdf_attribute_manager.py

```diff
@@ -0,0 +1,192 @@
+            const.CDF_EPOCH.value: "CDF_EPOCH",
+            const.CDF_EPOCH16.value: "CDF_EPOCH16",
+            const.CDF_TIME_TT2000.value: "CDF_TIME_TT2000",
+        }
+        self.numpytypedict = {
+            const.CDF_BYTE.value: np.int8,
+            const.CDF_CHAR.value: np.int8,
+            const.CDF_INT1.value: np.int8,
+            const.CDF_UCHAR.value: np.uint8,
+            const.CDF_UINT1.value: np.uint8,
+            const.CDF_INT2.value: np.int16,
+            const.CDF_UINT2.value: np.uint16,
+            const.CDF_INT4.value: np.int32,
+            const.CDF_UINT4.value: np.uint32,
+            const.CDF_INT8.value: np.int64,
+            const.CDF_FLOAT.value: np.float32,
+            const.CDF_REAL4.value: np.float32,
+            const.CDF_DOUBLE.value: np.float64,
+            const.CDF_REAL8.value: np.float64,
+            const.CDF_EPOCH.value: np.float64,
+            const.CDF_EPOCH16.value: np.dtype((np.float64, 2)),
+            const.CDF_TIME_TT2000.value: np.int64,
+        }
+        self.timetypes = [
+            const.CDF_EPOCH.value,
+            const.CDF_EPOCH16.value,
+            const.CDF_TIME_TT2000.value,
+        ]
+
+        # Load Default IMAP Global Attributes
+        self.global_attributes = self._load_yaml_data(
+            str(self.source_dir / DEFAULT_GLOBAL_CDF_ATTRS_FILE)
+        )
+        self.variable_attributes = dict()
+
+    def _load_default_global_attr_schema(self) -> dict:
+        default_schema_path = str(
+            self.source_dir
+            / "shared"
+            / DEFAULT_GLOBAL_CDF_ATTRS_SCHEMA_FILE
+        )
+        # Load the Schema
+        return self._load_yaml_data(default_schema_path)
+
+    def _load_default_variable_attr_schema(self) -> dict:
+        default_schema_path = str(
+            self.source_dir
+            / "shared"
+            / DEFAULT_VARIABLE_CDF_ATTRS_SCHEMA_FILE
+        )
+        # Load the Schema
+        return CdfAttributeManager._load_yaml_data(default_schema_path)
+
+    def load_global_attributes(self, file_path: str):
+        """
+        Update the global attributes property with the attributes from the file.
+
+        Parameters
+        ----------
+        file_path: str
+            file path to load, under self.source_dir.
+
+        """
+        self.global_attributes.update(
+            self._load_yaml_data(
+                str(self.source_dir / file_path)
+            )
+        )
+
+    @staticmethod
+    def _load_yaml_data(file_path: str | Path) -> dict:
+        """
+        Load a yaml file from the provided path.
+
+        Parameters
+        ----------
+        file_path: str | Path
+            path to the yaml file to load.
+
+        Returns
+        -------
+            Loaded yaml.
+        """
+        with open(file_path, "r") as file:
+            return yaml.safe_load(file)
+
+    def global_attribute_template(self, logical_source_id=None) -> OrderedDict:
```

Comment:
Is there a need for `OrderedDict`? Python dicts are now guaranteed ordered, so I find it easier to just use them if there isn't a specific reason for the `OrderedDict` implementation.
```suggestion
    def global_attribute_template(self, logical_source_id=None) -> dict:
```

Reply:
Are they really? I didn't realize that! Yeah, I'll remove ordereddict